public/app/features/explore/state/query.ts

```diff
@@ -166,6 +189,15 @@ export interface ClearCachePayload {
   exploreId: ExploreId;
 }
 export const clearCacheAction = createAction<ClearCachePayload>('explore/clearCache');
+
+/**
+ * Stores new value of auto-load logs volume switch. Used only internally. changeAutoLogsVolume() is used to
+ * update auto-load and load logs volume if it hasn't been loaded.
+ */
```

Comment:
Why are we storing auto-load logs volume at 2 places - local storage and redux store. Couldn't we just use the value from local storage?

Reply:
I think it makes sense to keep in redux store to decouple "how" the value is stored from components simply using the value. These components shouldn't be aware of that the value is stored in local storage, should just get the value.

But actually at the moment the value is stored and retrieved in two different components. I'll try to move it to the reducer.

Reply:
i was lurking on this and please don't use the reducer to do this, it shouldn't have any side effect.

we should create a middleware for it (but that would mean having it outside of explore's scope) or subscribe to the store somewhere in explore (maybe when mounting `Explore.ts`) and update the value in localstorage from the subscription, then when getting the initial state for explore we can read that value from localstorage. this second approach is maybe simpler but we should be careful with jsdom an tests

Reply:
- [x] Move `autoLoadLogsVolume` local storage handling (f9fddd46b7b08e725401d8d716791f01b2676f9c)

Moved handling up and added store subscription.